skbio/stats/composition.py

```diff
@@ -896,9 +961,27 @@ def ancom(table, grouping,
         else:
             nu = cutoff[4]
         reject = (W >= nu*n_feat)
-    labs = mat.columns
-    return pd.DataFrame({'W': pd.Series(W, index=labs),
-                         'reject': pd.Series(reject, index=labs)})
+
+    feat_ids = mat.columns
+    ancom_df = pd.DataFrame(
+        {'W': pd.Series(W, index=feat_ids),
+         'Reject null hypothesis': pd.Series(reject, index=feat_ids)})
+
+    if len(percentiles) == 0:
+        return ancom_df, pd.DataFrame()
+    else:
+        data = []
+        columns = []
+        for group in groups:
+            feat_dists = mat[grouping == group]
```

Comment:
This should be `mat[cats == group]` because `cats` is index-aligned to `mat` and `grouping` isn't. I don't think there's a bug here because pandas will index-align the boolean `Series` with `mat` before slicing, but we get a warning in that case:

```python
UserWarning: Boolean Series key will be reindexed to match DataFrame index.
```

Using `cats` should avoid this warning and an unnecessary alignment step.